src/middlewared/middlewared/etc_files/scst.conf.mako

```diff
@@ -67,35 +86,42 @@
                 if 'address' in addr:
                     failover_virtual_aliases.append(addr['address'])
 
+    cluster_mode_targets = []
+    cluster_mode_luns = {}
+    clustered_extents = set()
+    all_cluster_mode = False
     if failover_status == "MASTER":
-        middleware.call_sync("iscsi.target.logout_ha_targets")
         local_ip = middleware.call_sync("failover.local_ip")
         dlm_ready = middleware.call_sync("dlm.node_ready")
+        if alua_enabled:
+            clustered_extents = set(middleware.call_sync("iscsi.target.clustered_extents"))
+            cluster_mode_targets = middleware.call_sync("iscsi.target.cluster_mode_targets")
     elif failover_status == "BACKUP":
         if alua_enabled:
-            logged_in_targets = middleware.call_sync("iscsi.target.login_ha_targets")
-            cluster_mode_targets = middleware.call_sync('failover.call_remote', 'iscsi.target.cluster_mode_targets')
+            if middleware.call_sync("iscsi.alua.standby_write_empty_config"):
+                logged_in_targets = {}
+            else:
+                logged_in_targets = middleware.call_sync("iscsi.target.login_ha_targets")
+                try:
+                    (cluster_mode_targets, cluster_mode_luns) = middleware.call_sync('failover.call_remote', 'iscsi.target.cluster_mode_targets_luns')
+                except CallError as e:
+                    if e.errno != CallError.ENOMETHOD:
+                        raise FileShouldNotExist
```

Comment:
This means that if the other node is down for whatever reason, the entire scst.conf won't be generated since you're raising FileShouldNotExist. Is that desired?

Reply:
No, not desired.  Modified.